### DISCUSSION TITLE
How did you figure this out?

### DISCUSSION THREAD

**Discussion by kobbled** [gavanderhoorn/comet_rpc#4] [General]
Hey @gavanderhoorn, First off this looks like great work! I am just reading through it so I haven't tested it out anywhere, but just wanted to congratulate you on your discovery! I am mainly curious in how you figured this out? How did you find COMET then reverse engineer it? Are you planning on leveraging this with the ROS driver somehow?

What crazy lengths you have to go to to use a FANUC robot!

Cheers,

Matt

**Comment by gavanderhoorn** [comment]
> I am mainly curious in how you figured this out? How did you find COMET then reverse engineer it?

Just to clarify: I don't really consider this reverse engineering. I haven't looked at `COMET` itself, that wasn't needed.

If you open up any of the remote pendants, or iRProgrammer, they're "just" websites (and in quotes, as it's obviously a complex piece of software). If they load in a browser, these days, they must be either HTML + JS, WebAssembly or some other form of loadable 'plugin'. ActiveX doesn't work any more, no plugins are needed, so I figured it's likely HTML + JS.

After that it just took some sniffing with Wireshark.

> Are you planning on leveraging this with the ROS driver somehow?

Maybe.

For motion this isn't too interesting -- or at least, probably not for anything significantly better than what the driver does today. Might be nice to avoid the custom protocol, and offer (better/cleaner) support for IO and setting variables, etc.

But `R912` can also do all of that, so the main advantage this has is it doesn't appear to require any additional options.

> What crazy lengths you have to go to to use a FANUC robot!

Well to be fair: if you really just want to use *use* it as-is, it all just works -- typically. No need to spend time figuring out all this stuff.

I just happen to have strange / weird interests and/or requirements.
